lib/model-api/utils.rb

```diff
@@ -419,10 +419,49 @@ def process_collection_includes(collection, opts = {})
           includes << attr_metadata[:key] if attr_metadata[:type] == :association
         end
         includes = includes.compact.uniq
+        includes = remove_excluded_associations(includes, opts)
         collection = collection.includes(includes) if includes.present?
         collection
       end
 
+      # New Method: Does not eager load associations mentioned in the exclude_assocations
+      # array defined by user.
+      def remove_excluded_associations(includes, opts)
+        includes_dup = includes.compact.uniq.deep_dup
+        exclude_associations = opts[:exclude_associations]
+        return includes_dup unless exclude_associations.present?
+        exclude_associations.each do |association|
+          process_collection_array(includes_dup, association.to_sym)
+        end
+        includes_dup
+      end
+
+      # If the association is present in array form (in collection_includes) delete it.
+      # Then loop over the array elements to remove the association incase of nested associations.
+      def process_collection_array(array_collection, association)
```

Comment:
`process_collection_array` and `process_collection_hash` are very generic names. It's not clear what logic they perform. Perhaps something like `remove_association_from_array` and `remove_association_from_hash` would be more appropriate?